start.py

```diff
@@ -1,47 +1,230 @@
+        
+        self._value = piece_value
+
+        self._bin_value = [0, 0, 0, 0]
+        if piece_value >= 8: # color
+            self._bin_value[0] = 1
+            piece_value -= 8
+        if piece_value >= 4: # shape
+            self._bin_value[1] = 1
+            piece_value -= 4
+        if piece_value >= 2: # filling
+            self._bin_value[2] = 1
+            piece_value -= 2
+        self._bin_value[3] = piece_value # height
+
+    def is_empty(self):
+        return self._bin_value == None
+
+    def is_white(self):
+        return self._bin_value[0] == 1
+
+    def is_circle(self):
+        return self._bin_value[1] == 1
+
+    def is_square(self):
+        return self._bin_value[2] == 1
+
+    def is_high(self):
+        return self._bin_value[3] == 1
+
+    def decimal(self): # empty pieces return -1
+        return self._value
+
+    def binary(self): # empty pieces return None
+        return self._bin_value
+
+    def __str__(self):
+        return "%d%d\n%d%d" % (self._bin_value[0], self._bin_value[1], self._bin_value[2], self._bin_value[3])
+
+class Board:
+    def __init__(self): # creates board filled with empty pieces
+        self._board = [[Piece(None) for tile in range(4)] for row in range(4)]
+
+    def has_empty_tiles(self):
+        for row in self._board:
+            for tile in row:
+                if tile.is_empty():
+                    return True
+        return False
+
+    def is_avaliable(self, row, col):
+        return self._board[row - 1][col - 1].is_empty()
+
+    def place(self, piece, row, col):
+        self._board[row - 1][col - 1] = piece
+
+    def is_winning(self, piece): # Temporarly false
+        return False
+
+    def format(self): # returns board in string format "t t ... t t" (t - 16 tile values)
+        " ".join(" ".join(str(tile.decimal()) for tile in row) for row in _board)
+    
+    def __str__(self):
+        readable_board  = "#|1111|2222|3333|4444|\n"
+        readable_board += "#|----|----|----|----|\n"
+        
+        for m in range(4):
+            readable_row = "%d|" % (m+1)
+            for n in range(4):
+                piece = self._board[m][n]
+                if piece.is_empty():
+                    readable_row += "    |"
+                else:
+                    piece_binary = piece.binary()
+                    readable_row += " %d%d |" % (piece_binary[0], piece_binary[1])
+            readable_row += "\n"
+            readable_row += "%d|" % (m+1)
+            for n in range(4):
+                piece = self._board[m][n]
+                if piece.is_empty():
+                    readable_row += "    |"
+                else:
+                    piece_binary = piece.binary()
+                    readable_row += " %d%d |" % (piece_binary[2], piece_binary[3])
+            readable_row += "\n"
+            readable_row +=  "#|----|----|----|----|\n"
+            readable_board += readable_row
+
+        return readable_board
+
+class Game:
```

Comment:
dobrze by było zmienić nazwę tej klasy na QuartoGame, żeby było wiadomo co to za gra

Reply:
Pełna zgoda.